virtual-setup/working-with-your-own-data.md

```diff
@@ -0,0 +1,233 @@
+
+**Step 1)** Go to the Terminal tab in your RStudio session.
+
+![Terminal tab](screenshots/rstudio-session-terminal.png)
+
+**Step 2)** Set up your `wget` command in a script (or notebook) using this template.
+
+The most simple `wget` command just needs the URL to pull the file from.
+
+*Template:*
+```
+wget <URL>
+```
+
+*Specific example:* Here's an example of us downloading a file from ArrayExpress
+```
+wget https://www.ebi.ac.uk/arrayexpress/files/E-GEOD-67851/E-GEOD-67851.processed.1.zip
+```
+
+By default, the file will be saved to the current directory and the file name it had from its origin (so with the above example `E-GEOD-67851.processed.1.zip`).
+
+Likely you will want to be more specific about where you are saving the file to and what you are calling it.
+For that, we can use the `-O`, or `output` option with our `wget` command and specify a file path.  
+
+*Template:*
+```
+wget -O <FILE_PATH_TO_SAVE_TO> <URL>
+```
+
+*Specific example:* Here's an example where we will download that same array express file, but instead save it to the `data` folder and call it `some_array_data.zip`.
+(Best to keep the file extension consistent to avoid troubles!)
+
+```
+wget -O data/some_array_data.zip https://www.ebi.ac.uk/arrayexpress/files/E-GEOD-67851/E-GEOD-67851.processed.1.zip
+```
+
+`-O` is one of many `wget` command options.
+To see the complete list of `wget` options, use the command: `wget -h` in bash.
+
+[See more `wget` examples](https://www.tecmint.com/10-wget-command-examples-in-linux/)
+
+### Upload data from your own computer
+
+If the data you want to use is stored locally on your computer, here's how we recommend uploading it to the RStudio Server.
+
+**Step 1)** We recommend you compress your data folder into a single zip file.
+
+For most operating systems, you can right-click on your data folder, and choose `Compress` to zip up your files
+- [Windows zipping](https://support.microsoft.com/en-us/help/14200/windows-compress-uncompress-zip-files)
+- [Mac zipping](https://www.imore.com/how-compress-file-your-mac)
+
+For reference, here's how you [compress files from the command line](https://coolestguidesontheplanet.com/how-to-compress-and-uncompress-files-and-folders-in-os-x-lion-10-7-using-terminal/).
+
+**Step 2)** Once your data is compressed to a single file, [navigate to your RStudio session](./rstudio-login.md).
+
+**Step 3)** Use the `Upload button` to choose your compressed data folder.
+
+This button is in the lower right panel of your RStudio session:
+![Upload button](screenshots/upload-button.png)
+
+A mini screen will pop up asking you to choose the file you want to upload:
+
+![Choose file](screenshots/upload-choose-file.png)
+
+Choose your compressed data file, and click `OK`.
+This may take some time, particularly if you have a large dataset.  
+
+When the server is finished uploading your data, you should see your file in your `home` directory!
```

Comment:
Is it automatically uncompressed? If not, do we direct to do so?

Reply:
Yes. It's automatically uncompressed whether we like it or not.

Reply:
I can add that info on here so people are aware.

Reply:
Github does have an emoticon that captures my feelings about this information.